Source/GTMSessionFetcher.m

```diff
@@ -3728,10 +3734,60 @@ - (void)setCallbackQueue:(nullable dispatch_queue_t)queue {
   @synchronized(self) {
     GTMSessionMonitorSynchronized(self);
 
+    GTMSESSION_ASSERT_DEBUG(![self isFetchingUnsynchronized],
+                            @"callbackQueue should not change after beginFetch has been invoked");
+
     _callbackQueue = queue ?: dispatch_get_main_queue();
+    _serialCallbackQueue = nil;
+  }  // @synchronized(self)
+}
+
+/**
+ * The serial queue to which callbacks should be dispatched.
+ *
+ * Callbacks need to be dispatched serially. For example, the accumulate data block should never be
+ * run after the completion handler. Serial dispatch is guaranteed by dispatching to an intermediate
+ * serial queue that targets the actual callback queue (which may or may not be serial).
+ */
+- (nonnull dispatch_queue_t)serialCallbackQueue {
+  @synchronized(self) {
+    GTMSessionMonitorSynchronized(self);
+
+    return [self serialCallbackQueueUnsynchronized];
   }  // @synchronized(self)
 }
 
+- (nonnull dispatch_queue_t)serialCallbackQueueUnsynchronized {
+  GTMSessionCheckSynchronized(self);
+
+  if (!_serialCallbackQueue) {
+    _serialCallbackQueue = CreateSerialCallbackQueueFromCallbackQueue(_callbackQueue);
+  }
+  return _serialCallbackQueue;
+}
+
+static dispatch_queue_t CreateSerialCallbackQueueFromCallbackQueue(dispatch_queue_t callbackQueue) {
+  const char *callbackQueueLabel = dispatch_queue_get_label(callbackQueue);
+  NSString *serialCallbackQueueLabel;
+  if (callbackQueueLabel[0] == '\0') {
+    serialCallbackQueueLabel = @"GTMSessionFetcher serial queue targeting callback queue";
```

Comment:
Aren't queue labels supposed to be reverse dns?

Reply:
Which community is that convention from? Regardless, I think the convention should be broken in this case; not sure how we would express `[[NSString alloc] initWithFormat:@"GTMSessionFetcher serial queue targeting `%s`", callbackQueueLabel]` in reverse DNS given that the caller’s callback queue label may not use the reverse DNS convention.

Reply:
Apple documents it on the apis:

https://developer.apple.com/documentation/dispatch/1453030-dispatch_queue_create
https://developer.apple.com/documentation/dispatch/1642205-dispatch_queue_create_with_targe

> Because applications, libraries, and frameworks can all create their own dispatch queues, a reverse-DNS naming style (_com.example.myqueue_) is recommended. This parameter is optional and can be NULL.

Reply:
It's Apple's recommendation. I've submitted a PR to fix this, and a potential invalid access bug since the code doesn't check the return value of `dispatch_queue_get_label` for `NULL`.

Reply:
Okie, thanks! Re `NULL`, see https://github.com/google/gtm-session-fetcher/pull/230#discussion_r668348805.